File: examples/03_module/order/order.js
// Lean Code — BSD 3-Clause License — Vivian Voss, 2026
// Scope: Read orders from file.

import { readFileSync } from "node:fs";

function readOrderFromFile(path) {
  const lines = readFileSync(path, "utf-8").trim().split("\n");

  return {
    id: parseInt(lines[0], 10),
    product: lines[1],
    quantity: parseInt(lines[2], 10),
  };
}

export { readOrderFromFile };

File: examples/03_module/order/order.py
# Lean Code — BSD 3-Clause License — Vivian Voss, 2026
# Scope: Read orders from file.


def read_order_from_file(path):
    lines = open(path).read().strip().split("\n")

    return {
        "id": int(lines[0]),
        "product": lines[1],
        "quantity": int(lines[2]),
    }
